course2/lecture_en.md

```diff
@@ -0,0 +1,379 @@
+
+    tmp
+  }
+  tmp + another_tmp + y
+}
+```
+
+First, we can replace all occurrences of `x` and `y` with their values, thereby omitting their variable bindings.
+
+```moonbit expr
+// Omit the variable binding of y
+let value = {
+  // Omit the variable binding of x
+  let tmp = 1 * 2 // Replace x
+  let another_tmp = {
+    let tmp = 1 * 3 // Replace x
+
+    tmp
+  }
+  tmp + another_tmp + 10 // Replace y
+}
+```
+
+Then, we can reduce the expression for two variable bindings of `tmp`, and replace the occurrences of `tmp` in the expression block of the variable binding of `another_tmp`.
+
+```moonbit expr
+let value = {
+  let tmp = 2 // Reduce the expression on the right-hand side
+  let another_tmp = {
+    let tmp = 3 // Reduce the expression on the right-hand side
+
+    3 // Replace tmp
+  }
+  tmp + another_tmp + 10
+}
+```
+
+After that, we can now compute the value of `another_tmp`, which is determined by the last expression in the expression block.
+
+```moonbit expr
+let value = {
+
+  let tmp = 2
+  let another_tmp = 3 // Reduce the expression on the right-hand side
+  tmp + another_tmp + 10
+}
+```
+
+Thus, the remaining occurrences of identifiers in the expression block of `value` can also be replaced with their values.
+
+```moonbit expr
+let value = {
+
+  let tmp = 2
+  let another_tmp = 3
+  2 + 3 + 10
+}
+```
+
+Fanally, we get the value of `value`.
+
+```moonbit expr
+let value = 15
+```
+
+#### Conditional Expression
+
+Conditional expressions enable one to obtain different values ​​based on specified logical conditions.
+
+In MoonBit, its syntax is:
+
+```
+if condition
+  expression block｜if condition is true
+else
+  expression block｜if condition is false
+```
+
+In MoonBit, conditional expressions are also expressions and can be used within other expressions. For example,
+
+- `( if 1 < 100 { 1 } else { 0 } ) * 10`
+- `( if x > y { "x" } else { "y" } ) + " is bigger"`
+- `if 0.1 + 0.2 == 0.3 { "Great!" } else { "C'est la vie :-)" }`
+
+The expression blocks in the branches must have the same type, and the type of the entire conditional expression is determined by on the type of the expression blocks in the branches. The type of the condition must be a Boolean.
+
+![](../pics/if-else-then.drawio.png)
+
+In the above conditional expression, the condition is a `Bool` expression to check the equality of two floating-point numbers; the values ​​of the two branches are both of type `String`, so the entire expression is also of type `String`.
+
+The value of a conditional expression depends on the reduction result of the condition being true or false. For example,
+
+$$\begin{align}
+    & \texttt{if 1 < 100 \{ 1 \} else \{ 0 \} * 10} \\
+    \mapsto & \texttt{if true \{ 1 \} else \{ 0 \} * 10} \\
+    \mapsto & \texttt{1 * 10} \\
+    \mapsto & \texttt{10}
+\end{align}$$
+
+A conditional expression must have two branches. Otherwise, if the condition is false, what should be the value of the expression?
```

Comment:
It would be more accurate to introduce `Unit` and say that we can omit the other branch if the type is `Unit` (as if there's an implicit branch returning `()`) (I know that Chinese version didn't have this part)